sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java

```diff
@@ -688,4 +721,513 @@ Mono<Response<BinaryData>> getChatCompletionsWithAzureExtensionsWithResponse(
+        acceptedFormats.add(AudioTranscriptionFormat.VTT);
+        acceptedFormats.add(AudioTranscriptionFormat.SRT);
+        if (!acceptedFormats.contains(audioTranscriptionOptions.getResponseFormat())) {
+            return monoError(LOGGER,
+                    new IllegalArgumentException("This operation does not support the requested audio format"));
+        }
+        // embedding the `model` in the request for non-Azure case
+        if (this.openAIServiceClient != null) {
+            audioTranscriptionOptions.setModel(deploymentOrModelName);
+        }
+        MultipartDataHelper helper = new MultipartDataHelper();
+        MultipartDataSerializationResult result = helper.serializeRequest(audioTranscriptionOptions, fileName);
+
+        String multipartBoundary = helper.getBoundary();
+        RequestOptions requestOptions = new RequestOptions();
+        requestOptions
+                .setHeader(HttpHeaderName.CONTENT_TYPE, "multipart/form-data;" + " boundary=" + multipartBoundary)
+                .setHeader(HttpHeaderName.CONTENT_LENGTH, String.valueOf(result.getDataLength()));
+
+        Mono<Response<BinaryData>> response =
+                openAIServiceClient != null
+                        ? this.openAIServiceClient.getAudioTranscriptionAsPlainTextWithResponseAsync(
+                        deploymentOrModelName, result.getData(), requestOptions)
+                        : this.serviceClient.getAudioTranscriptionAsPlainTextWithResponseAsync(
+                        deploymentOrModelName, result.getData(), requestOptions);
+        return response.map(binaryData -> binaryData.getValue().toString());
+    }
+
+    /**
+     * Gets the transcription of the translation of the audio file passed with additional metadata
+     *
+     * @param deploymentOrModelName Specifies either the model deployment name (when using Azure OpenAI) or model name
+     * (when using non-Azure OpenAI) to use for this request.
+     * @param audioTranslationOptions audio translation request object
+     * @param fileName the name of the file that is represented in your "file" field of {@link AudioTranscriptionOptions}
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return {@link AudioTranscription} transcribed text and associated metadata of the English translation from provided spoken
+     * audio data in a structure.
+     */
+    public Mono<AudioTranscription> getAudioTranslation(
```

Comment:
re-order the method parameters to be "String deploymentOrModelName,  String fileName, AudioTranslationOptions audioTranslationOptions", as `fileName` is a required parameter